### PR TITLE
v2 Workflow spec api returns the specifications in pre 2.8 format

### PR DESCRIPTION
**Pull request by sagarkapare**
Jira - https://issues.cask.co/browse/CDAP-1669

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java [hunk 1]
```diff
@@ -756,9 +773,62 @@ public void sparkSpecification(HttpRequest request, HttpResponder responder,
   public void workflowSpecification(HttpRequest request, HttpResponder responder,
                                     @PathParam("app-id") final String appId,
                                     @PathParam("workflow-id")final String workflowId) {
-    programLifecycleHttpHandler.programSpecification(RESTMigrationUtils.rewriteV2RequestToV3(request), responder,
```
Comment (nitinmotgi) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java]: Why are we diverging from centralizing this into a programSpecification ?

Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java]: We changed WorkflowSpecifications in 2.8. However we still want V2 WorkflowSpecification api to return the data in the format which is compatible to 2.7, which is why special handling is needed here.

**Comment by sreevatsanraman** [comment]
LGTM :+1:

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-RBT170

**Pull request merged**